fixtures/node-app-pages/tests/index.test.ts

```diff
@@ -1,36 +1,46 @@
 import { fork } from "child_process";
 import path from "path";
 import { fetch } from "undici";
-import type { ChildProcess } from "child_process";
 
 describe("Pages Dev", () => {
-	let wranglerProcess: ChildProcess;
-	let ip: string;
-	let port: number;
-	let resolveReadyPromise: (value: unknown) => void;
-	const readyPromise = new Promise((resolve) => {
-		resolveReadyPromise = resolve;
-	});
+	it.concurrent(
+		"should work with `--node-compat` when running code requiring polyfills",
+		async () => {
+			const { ip, port, exit } = await runWranglerPagesDev(
+				"public",
+				"--node-compat",
+				"--port=0"
+			);
+			const response = await fetch(`http://${ip}:${port}/stripe`);
 
-	beforeAll(() => {
-		wranglerProcess = fork(
-			path.join("..", "..", "packages", "wrangler", "bin", "wrangler.js"),
-			["pages", "dev", "public", "--node-compat", "--port=0"],
-			{
-				stdio: ["inherit", "inherit", "inherit", "ipc"],
-				cwd: path.resolve(__dirname, ".."),
-			}
-		).on("message", (message) => {
-			const parsedMessage = JSON.parse(message.toString());
-			ip = parsedMessage.ip;
-			port = parsedMessage.port;
-			resolveReadyPromise(undefined);
-		});
-	});
+			await expect(response.text()).resolves.toContain(
+				`"PATH":"path/to/some-file","STRIPE_OBJECT"`
+			);
 
-	afterAll(async () => {
-		await readyPromise;
-		await new Promise((resolve, reject) => {
+			await exit();
+		},
+		10000
+	);
+});
+
+async function runWranglerPagesDev(publicPath: string, ...options: string[]) {
+	let resolveReadyPromise: (value: { ip: string; port: number }) => void;
+	const ready = new Promise<{ ip: string; port: number }>(
+		(resolve) => (resolveReadyPromise = resolve)
+	);
+	const wranglerProcess = fork(
+		"../../packages/wrangler/bin/wrangler.js",
+		["pages", "dev", publicPath, ...options],
+		{
+			stdio: ["ignore", "ignore", "ignore", "ipc"],
+			cwd: path.resolve(__dirname, ".."),
+		}
+	).on("message", (message) => {
+		resolveReadyPromise(JSON.parse(message.toString()));
+	});
+	async function exit() {
+		await ready;
```

Comment:
You'll have guaranteed that `ready` is already awaited here on L54, so I don't think it's necessary here.

It was needed previously because of this bug in Jest (https://github.com/facebook/jest/issues/7997) which meant that sometimes a `afterAll()` actually executed before any of the tests.

Reply:
I think we can avoid concurrent for these tests. It doesn't seem to make any real difference to the speed or reliability.